### PR TITLE
chore: refactoring market

### PR DESCRIPTION
**Pull request by quasisamurai**
- refactored Bill function.

- Now when you close deal - market wouldn't reserve your funds and return it back instantly.

- fixed some event params (not critical).

- refactored refunding when deal closes.

### CODE REVIEW

File: blockchain/source/contracts/Market.sol [hunk 1]
```diff
@@ -699,6 +633,89 @@ contract Market is Ownable {
...
+            uint nextPeriodSum = CalculatePayment(deal.price, nextPeriod).sub(deals[dealID].blockedBalance);
+
+            if (token.balanceOf(deal.consumerID) >= nextPeriodSum) {
+                require(token.transferFrom(deal.consumerID, this, nextPeriodSum));
+                deals[dealID].blockedBalance = deals[dealID].blockedBalance.add(nextPeriodSum);
+            } else {
+                emit Billed(dealID, deals[dealID].blockedBalance);
+                InternalCloseDeal(dealID);
+                RefundRemainingFunds(dealID);
+                deals[dealID].blockedBalance = 0;
```
Comment (sokel) [inline blockchain/source/contracts/Market.sol]: may be do it in `RefundRemainingFunds`?

File: blockchain/source/contracts/Market.sol [hunk 2]
```diff
@@ -699,6 +633,89 @@ contract Market is Ownable {
     }
     // INTERNAL
 
+    function InteralBill(uint dealID) internal returns (bool){
+        require(deals[dealID].status == DealStatus.STATUS_ACCEPTED);
```
Comment (sokel) [inline blockchain/source/contracts/Market.sol]: its more applicable at `Bill`

File: blockchain/source/contracts/Market.sol [hunk 3]
```diff
@@ -343,86 +343,20 @@ contract Market is Ownable {
         }
 
         AddToBlacklist(dealID, blacklisted);
-        Bill(dealID);
+        InteralBill(dealID);
```
Comment (antmat) [inline blockchain/source/contracts/Market.sol]: typo

File: blockchain/source/contracts/Market.sol [hunk 4]
```diff
@@ -699,6 +633,89 @@ contract Market is Ownable {
     }
     // INTERNAL
 
+    function InteralBill(uint dealID) internal returns (bool){
+        require(deals[dealID].status == DealStatus.STATUS_ACCEPTED);
+        require(msg.sender == deals[dealID].supplierID || msg.sender == deals[dealID].consumerID || msg.sender == deals[dealID].masterID);
```
Comment (sokel) [inline blockchain/source/contracts/Market.sol]: this too

File: blockchain/source/contracts/Market.sol [hunk 5]
```diff
@@ -699,6 +633,89 @@ contract Market is Ownable {
...
+
+            if (token.balanceOf(deal.consumerID) >= nextPeriodSum) {
+                require(token.transferFrom(deal.consumerID, this, nextPeriodSum));
+                deals[dealID].blockedBalance = deals[dealID].blockedBalance.add(nextPeriodSum);
+            } else {
+                emit Billed(dealID, deals[dealID].blockedBalance);
+                InternalCloseDeal(dealID);
+                RefundRemainingFunds(dealID);
+                deals[dealID].blockedBalance = 0;
+                return true;
```
Comment (sokel) [inline blockchain/source/contracts/Market.sol]: minor: this is unnecessary statement

File: blockchain/source/contracts/Market.sol (hunk 2 above)
Comment (antmat) [reply blockchain/source/contracts/Market.sol]: why? Is it valid to run IntarnalBill on closed deal?

File: blockchain/source/contracts/Market.sol (hunk 1 above)
Comment (antmat) [reply blockchain/source/contracts/Market.sol]: +

File: blockchain/source/contracts/Market.sol [hunk 6]
```diff
@@ -343,86 +343,20 @@ contract Market is Ownable {
...
-                deals[dealID].blockedBalance = deals[dealID].blockedBalance.add(nextPeriodSum);
-            } else {
-                emit Billed(dealID, deal.blockedBalance);
-                InternalCloseDeal(dealID);
-                require(token.transfer(deal.consumerID, deals[dealID].blockedBalance));
-                deals[dealID].blockedBalance = 0;
-                return true;
-            }
-        }
+        InteralBill(dealID);
```
Comment (3Hren) [inline blockchain/source/contracts/Market.sol]: Typo: s/Interal/Internal/

**Pull request merged**